spec/domain/accounts/users_spec.exs

```diff
@@ -39,26 +39,91 @@ defmodule Squadster.Domain.Accounts.UsersSpec do
   describe "auth_changeset" do
     context "when params are valid" do
       it "is valid" do
-        %{valid?: is_valid} = %{uid: "123", first_name: "John", last_name: "Galt"} |> User.changeset
+        %{valid?: is_valid} = %{uid: "123", first_name: "John", last_name: "Galt"} |> User.auth_changeset
         expect is_valid |> to(be true)
       end
-    end
 
-    context "when params are invalid" do
-      context "when first_name is not set" do
-        it "is not valid" do
-          %{valid?: is_valid} = %{uid: "123", last_name: "Galt"} |> User.auth_changeset
-          expect is_valid |> to(be false)
+      defmodule SimpleUserParam do
+        use ESpec, shared: true
+
+        let_overridable [:param, :default_value, :new_value]
+
+        context "when the param is already present on user" do
+          let! :user, do: insert(:user, "#{param()}": default_value())
+
+          it "should be execluded from changeset" do
+            changeset = user() |> User.auth_changeset(%{"#{param()}": new_value()})
+            expect changeset.changes[param()] |> to(eq nil)
+          end
+        end
+
+        context "when the param is not present on user" do
+          let! :user, do: insert(:user, "#{param()}": nil)
+
+          it "should not be execluded from changeset" do
+            changeset = user() |> User.auth_changeset(%{"#{param()}": new_value()})
+            expect changeset.changes[param()] |> to(eq new_value())
+          end
         end
       end
 
-      context "when last_name is not set" do
-        it "is not valid" do
-          %{valid?: is_valid} = %{uid: "123", first_name: "John"} |> User.auth_changeset
-          expect is_valid |> to(be false)
+      defmodule VkPriorityUserField do
+        use ESpec, shared: true
+
+        let_overridable [:param, :default_value, :new_value]
+
+        context "when the param is already present on user" do
+          let! :user, do: insert(:user, "#{param()}": default_value())
+
+          it "should not be execluded from changeset" do
+            changeset = user() |> User.auth_changeset(%{"#{param()}": new_value()})
+            expect changeset.changes[param()] |> to(eq new_value())
+          end
+        end
+
+        context "when the param is not present on user" do
+          let! :user, do: insert(:user, "#{param()}": nil)
+
+          it "should not be execluded from changeset" do
+            changeset = user() |> User.auth_changeset(%{"#{param()}": new_value()})
+            expect changeset.changes[param()] |> to(eq new_value())
+          end
         end
       end
 
+      it_behaves_like SimpleUserParam, param: :first_name, default_value: "John", new_value: "Jack"
```

Comment:
Why you didn't use `Faker`?

Reply:
Because it's just a test, not a factory or seed, we don't really need this data to be different from run to run, it will just reduce readability and stability

Reply:
Well, seems good